core/cmd/shell_local.go

```diff
@@ -900,6 +923,45 @@ func (s *Shell) CreateMigration(c *cli.Context) error {
 	return nil
 }
 
+// CleanupChainTables deletes database table rows based on chain type and chain id input.
+func (s *Shell) CleanupChainTables(c *cli.Context) error {
+	cfg := s.Config.Database()
+	parsed := cfg.URL()
+	if parsed.String() == "" {
+		return s.errorOut(errDBURLMissing)
+	}
+
+	dbname := parsed.Path[1:]
+	if !c.Bool("dangerWillRobinson") && !strings.HasSuffix(dbname, "_test") {
+		return s.errorOut(fmt.Errorf("Cannot reset database named `%s`. This command can only be run against databases with a name that ends in `_test`, to prevent accidental data loss. If you really want to delete chain specific data from this database, pass in the -dangerWillRobinson option", dbname))
+	}
+
+	db, err := newConnection(cfg)
+	if err != nil {
+		return s.errorOut(errors.Wrap(err, "Error connecting to the database"))
+	}
+
+	defer db.Close()
+
+	// Delete rows from each table based on the chain_id.
+	if strings.EqualFold("EVM", c.String("chaintype")) {
+		tables, err := getTablesContainingColumn(db, "evm_chain_id")
+		if err != nil {
+			return s.errorOut(errors.Wrap(err, "Failed to get tables"))
+		}
+		for _, tableName := range tables {
+			query := fmt.Sprintf("DELETE FROM %s WHERE evm_chain_id = $1", tableName)
+			_, err := db.Exec(query, c.Int("chainid"))
+			if err != nil {
+				fmt.Printf("Error deleting rows from %s: %v\n", tableName, err)
+			} else {
+				fmt.Printf("Rows with chain_id %d deleted from %s.\n", c.Int("chainid"), tableName)
+			}
+		}
+	}
```

Comment:
Else, return error for unknown chaintype?

Reply:
opps, lost it while refactoring